pkg/provisioner/ironic/delete_test.go

```diff
@@ -143,6 +143,19 @@ func TestDelete(t *testing.T) {
 				Value: true,
 			},
 		},
+		{
+			name: "power-change-with-locked-host",
+			ironic: testserver.NewIronic(t).Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+					PowerState:     powerOn,
+				},
+			).WithNodeStatesPower(nodeUUID, http.StatusConflict).WithNodeStatesPowerUpdate(nodeUUID, http.StatusConflict),
+			expectedDirty:        false,
```

Comment:
These expected values are consistent with what is returned by transientError(). http.StatusConflict is not resulting in HostLockedError.